File: week-04/day-01/sharpie.py
# Create Sharpie class
# We should know about each sharpie their color (which should be a string), width (which will be a floating point number), ink_amount (another floating point number)
# When creating one, we need to specify the color and the width
# Every sharpie is created with a default 100 as ink_amount
# We can use() the sharpie objects
# which decreases inkAmount

class Sharpie(object):
    ink_amount = float(100)


    def __init__(self, color, width):
        self.color = color
        self.width = float(width)
        

    def use(self):
        self.ink_amount -= 1
        

sharpie1 = Sharpie("blue", 3.5)
sharpie1.use()
print(sharpie1.color, sharpie1.width, sharpie1.ink_amount)

File: week-08/day-01/sharpie.js
// We should know about each sharpie:
// color (which should be a string)
// width (which will be a number)
// inkAmount (another number)
// When instantiating a Sharpie, we need to specify the color and the width
// Every sharpie is created with a default 100 as inkAmount
// We can use() the sharpie objects
// which decreases inkAmount by the width
// Write a loop that consumes all the sharpie's ink, but don't hardcode the required iteration count to your code

class Sharpie {
    
      constructor(color, width) {
        this.color = color;
        this.width = width;
        this.inkAmount = 100;
      }
    
      use() {
        this.inkAmount -= this.width;
      }
    }

    var redSharpie = new Sharpie('red', 0.5);
    
    while (redSharpie.inkAmount > 0) {
        redSharpie.use();
    }

    console.log(redSharpie.inkAmount);




    